Common/cpp/SharedItems/Shareables.h

```diff
@@ -285,7 +285,14 @@ class ShareableRemoteFunction
         function_(std::move(function)) {}
   jsi::Value toJSValue(jsi::Runtime &rt) override {
     if (runtimeHelper_->isUIRuntime(rt)) {
+#ifdef DEBUG
```

Comment:
Why is it only available in debug?

Reply:
It will throw "Object is not a function" in production and this way it works faster in release mode as it does not perform additional JSI call just for the sake of displaying an error message which only happens if you mess something up. I think it is the right tradeoff and I don't think it is likely that someone will only hit the error in release but never in production. I think the effort should be better spend on figuring out a solution for providing better stack traces which should give a much better visibility into the error root cause

Reply:
In general, I don't like such differences between in the behaviour of debug and release builds but this tradeoff sounds reasonable. 👍

BTW We could also analogously wrap `if (fun.__remoteFunction) { ... }` inside `if (__DEV__)` check so it can be optimized away but maybe it's not worth it.